### ISSUE TITLE
'Download all Tags' on Tag Management page does not download all data

### ISSUE THREAD

**Issue by eageldart** [MotusDev/Motus-TO-DO#1242]
Lately, when I click 'Download all Tags' on the Tag Management page, an excel spreadsheet downloads but does not contain all of the data. 

![image](https://user-images.githubusercontent.com/94930170/185954992-ad695da6-9c75-497a-89bf-3ef3fecccf89.png)

I have attached the excel file I got from project 72: 
[MotusTagsTemplate-72.csv](https://github.com/MotusDev/Motus-TO-DO/files/9395508/MotusTagsTemplate-72.csv)

And project 364 gave me data for 1 out of 16 tags.

**Comment by cjardine-bsc** [comment]
Its working correctly from here: https://motus.org/data/project/tags/template?projectID=72 

@Hussell  can you check what process the button on https://motus.org/data/project/tags?projectID=72 is calling and change it to the same as the button (tag deployments) on https://motus.org/data/project/tags/template?projectID=72

**Comment by bcreightonyoung** [comment]
This problem was actually quite a bit more subtle then changing the button. There was some stuff going on in the background with the filters (all, pending, active, terminated). The cause of this issue was that the "all" filter was being treated as "pending" and hence returning too few tags.

There is a bit of a separate issue here, which is that the count of tags for each filter (e.g. All (18)) on the webpage does not necessarily agree with the number of rows in the returned csv file. This is because the webpage is pulling data from the tags table and the csv is pulling data from the api, which is using vwBase_API_tag_deployments_template.

There is potentially some optimization to be done with how the data is populated on this page. 

Nonetheless, @eageldart see if you are getting a better (if not perfect) response from the download buttons on sandbox.

**Comment by eageldart** [comment]
I definitely get a better response than before. But the numbers are still a few shy less than what the Tab on the Tag Management counts.

**Comment by bcreightonyoung** [comment]
Yeah, there is definitely some more stuff to figure out with this one. I will have to figure out the process for the template data a little better before I can resolve this fully. We can keep this issue open until it's properly sorted out.
